tests/unit/data-access/MwApiLexemeCreator.test.ts

```diff
@@ -36,7 +38,64 @@ describe( 'MwApiLexemeCreator', () => {
 				lexicalCategory: 'Q456',
 			} ),
 			assert: 'user',
+			errorformat: 'html',
+			formatversion: 2,
 		} );
 	} );
 
+	describe( 'error handling', () => {
+
+		it( 'maps API errors', () => {
+			const deferred = $.Deferred();
```

Comment:
I'm a little confused as to why we need to use jQuery deferred objects here? Can't this be done with promises?

Reply:
Pushed a commit that uses a mock object instead, we can decide if we like that better or not. (I’d say it turned out less terrible than I feared.)